### PR TITLE
Add pull_request event to github actions hooks

### PR DESCRIPTION
**Pull request by k-nasa**
https://github.com/async-rs/async-std/pull/253

I forgot to add pull_request event hooks.
Probably necessary, but what about?

### CODE REVIEW

**Comment by k-nasa** [comment]
GitHub actions worked. Would have been necessary

スクリーンショット 2019-10-01 10 50 28

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-on: [push]
+on: [push, pull_request]
```
Comment (yoshuawuyts) [inline .github/workflows/ci.yml]: This should probably only be on `pull_request`. Else you get a double event triggering for each commit, which can be really noisy.

File: .github/workflows/clippy.yml [hunk 2]
```diff
@@ -1,4 +1,5 @@
-on: push
+on: [push, pull_request]
```
Comment (yoshuawuyts) [inline .github/workflows/clippy.yml]: This should probably only be on `pull_request`. Else you get a double event triggering for each commit, which can be really noisy.

**Comment by k-nasa** [comment]
Ok. I fixed. 
However, is ci not working when pushed to master branch?

**Comment by yoshuawuyts** [comment]
> However, is ci not working when pushed to master branch?

Not sure; in general all of our commits are through PRs anyway, and if Bors works then we don't need to run the *extra* CI run on master merges.

It looks like Clippy is failing though; not sure what's going on as it's a lot of JSON output.

**Comment by k-nasa** [comment]
@yoshuawuyts  My imagination is that this access token is mine (the person who pushed it) and I don't have write access, so it may have failed.
```yml
token: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by yoshuawuyts** [comment]
Alright then; let's merge this and see if it works :crossed_fingers:

**Pull request merged**